Wikipedia/Code/MWKSite.m

```diff
@@ -37,7 +37,7 @@ - (instancetype)initWithLanguage:(NSString*)language {
 }
 
 - (MWKSite* __nullable)initWithURL:(NSURL* __nonnull)url {
-    NSArray* hostComponents = [url.host componentsSeparatedByString:@"."];
+    NSMutableArray* hostComponents = [[url.host componentsSeparatedByString:@"."] mutableCopy];
```

Comment:
answered on hangout, but for posterity:

The code below this method doesn't work with mobile URLs and was the source of a bug where mobile URLs were never parsed properly.